azure/scope/identity.go

```diff
@@ -165,17 +157,46 @@ func (p *AzureCredentialsProvider) GetTokenCredential(ctx context.Context, resou
 		}
 		cred, authErr = NewWorkloadIdentityCredential(azwiCredOptions)
 
-	case infrav1.ServicePrincipal, infrav1.ServicePrincipalCertificate, infrav1.UserAssignedMSI:
-		if err := createAzureIdentityWithBindings(ctx, p.Identity, resourceManagerEndpoint, activeDirectoryEndpoint, clusterMeta, p.Client); err != nil {
-			return nil, err
+	case infrav1.ServicePrincipalCertificate:
+		// TODO: this is wrong. How to get the certificate data?
```

Comment:
the certificate and password should be in the k8s Secret no? https://capz.sigs.k8s.io/topics/multitenancy#aad-pod-identity-using-service-principal-with-certificate-deprecated

Reply:
Yes. Thanks for the assist!